models/stg_facebook_ads__basic_ad.sql

```diff
@@ -36,9 +36,29 @@ final as (
         cast(account_id as {{ dbt.type_bigint() }}) as account_id,
         impressions,
         coalesce(inline_link_clicks,0) as clicks,
-        spend,
-        reach,
-        frequency
+        spend
+
+        {# 
+            Reach and Frequency are not included in downstream models by default, though they are included in the staging model.
+            The below ensures that users can add Reach and Frequency to downstream models with the `facebook_ads__basic_ad_passthrough_metrics` variable
+            while avoiding duplicate column errors.
+        #}
+        {%- set check = [] %}
+        {%- for field in var('facebook_ads__basic_ad_passthrough_metrics') -%}
+            {%- if (field.alias if field.alias else field.name)|lower == 'reach' %}
+                {% do check.append('reach') %}
+            {% endif -%}
+            {%- if (field.alias if field.alias else field.name)|lower == 'frequency' %}
+                {% do check.append('frequency') %}
+            {% endif -%}
+        {% endfor -%}
+
+        {% if 'reach' not in check %}
+        , reach
+        {% endif %}
+        {% if 'frequency' not in check %}
+        , frequency
+        {%- endif %}
```

Comment:
It seems like we are doing a lot of hard conditional checks. I wonder if there is a way to make this more dynamic and less reliant on the hard coded conditional checks. 🤔 Similar to the approach you used in the macro.

Reply:
I just tested the following and it looks to achieve the same result without the hard conditional checks and less code. What are your thoughts?
```suggestion
        {%- set check = [] %}
        {%- for field in var('facebook_ads__basic_ad_passthrough_metrics') -%}
            {%- set field_name = field.alias|default(field.name)|lower %}
            {% if field_name in ['reach', 'frequency'] %}
                {% do check.append(field_name) %}
            {% endif %}
        {%- endfor %}

        {%- for metric in ['reach', 'frequency'] -%}
            {% if metric not in check %}
                , {{ metric }}
            {% endif %}
        {%- endfor %}
```